test/Response/SapiStreamEmitterTest.php

```diff
@@ -211,70 +445,80 @@ function () use (& $closureTrackMemoryUsage) {
-            $position = $args[0];
-            return true;
-        });
+    public function emitJsonResponseProvider()
+    {
+        return [[0.1                                                                         ],
+                ['test'                                                                      ],
+                [true                                                                        ],
+                [1                                                                           ],
+                [['key1' => 'value1']                                                        ],
+                [null                                                                        ],
+                [[[0.1, 0.2], ['test', 'test2'], [true, false], ['key1' => 'value1'], [null]]],
+        ];
+    }
 
-        $stream->eof()->will(function () use (& $contents, & $position) {
-            return ! isset($contents[$position]);
-        });
+    /**
+     * @dataProvider emitJsonResponseProvider
+     */
+    public function testEmitJsonResponse($contents)
+    {
+        $response = (new JsonResponse($contents))
+            ->withStatus(200);
 
-        $stream->tell()->will(function () use (& $position) {
-            return $position;
-        });
+        ob_start();
+        $this->emitter->emit($response);
+        $this->assertEquals('application/json', $response->getHeaderLine('content-type'));
+        $this->assertEquals(json_encode($contents), ob_get_clean());
+    }
 
-        $stream->read(Argument::type('integer'))->will(function ($args) use (& $contents, & $position) {
-            $data = substr($contents, $position, $args[0]);
-            $position += strlen($data);
-            return $data;
-        });
+    public function testEmitTextResponse()
+    {
+        $contents = 'Hello world';
 
-        $response = (new Response())
-            ->withStatus(200)
-            ->withHeader('Content-Range', "$unit $first-$last/$length")
-            ->withBody($stream->reveal());
+        $response = (new TextResponse($contents))
+            ->withStatus(200);
 
         ob_start();
-        $this->emitter->emit($response, $maxBufferLength);
-        $stream->seek(Argument::type('integer'), Argument::any())->shouldBeCalledTimes($seekable ? 1 : 0);
-        $stream->read(Argument::type('integer'))->shouldBeCalledTimes($expectedReads);
-        $this->assertEquals(substr($contents, $first, $last - $first + 1), ob_get_clean());
+        $this->emitter->emit($response);
+        $this->assertEquals('text/plain; charset=utf-8', $response->getHeaderLine('content-type'));
+        $this->assertEquals($contents, ob_get_clean());
     }
 
     public function contentRangeProvider()
     {
         return [
-            ['bytes 0-2/*', 'Hello world', 'Hel'],
-            ['bytes 3-6/*', 'Hello world', 'lo w'],
-            ['items 0-0/1', 'Hello world', 'Hello world'],
+                ['bytes 0-2/*', 'Hello world', 'Hel'],
```

Comment:
Okay. It was supposed to be just a reordering of this piece of the code. I messed things.